src/Pact/Compile.hs

```diff
@@ -214,28 +213,37 @@ defconst = do
   modName <- currentModule'
   a <- arg
   v <- term
-  m <- meta
-  when (isJust (m ^. mModel)) $ syntaxError "@model not permitted on defconst"
+
+  m <- meta ModelNotAllowed
   TConst a modName (CVRaw v) m <$> contextInfo
 
-meta :: Compile Meta
-meta = atPairs <|> try docStr <|> return def
+data ModelAllowed
+  = ModelAllowed
+  | ModelNotAllowed
+
+meta :: ModelAllowed -> Compile Meta
```

Comment:
Instead of just one "meta" and the ModelAllowed boolean, prefer we just use different combinators, where `meta` captures both, and `docOnlyMeta` only recognizes `@doc`

Reply:
I just added a commit that does this -- https://github.com/kadena-io/pact/pull/278/commits/1ee878b4b607b68d673dc8012457f0e12e798b40. Unfortunately it introduces some duplication because even in the case where we don't allow `@model` we want to parse it and explicitly error for a better error message. So I think I'd prefer to not apply this commit.

Reply:
I don't see the point of spending parser cycles trying something that is illegal in the first place. A simple "@doc or docstring" parser is fast and simple, and if sharing code is important it's easy to factor accordingly.

Reply:
Any comment here? To restate/elaborate:
`metaNoModel` (for which I think `docOnlyMeta` is better, to avoid negatives in names) should simply try the two alternatives and fail, and not parse things that are illegal, which is slower and inconsistent with other combinators here.

```haskell
docOnlyMeta :: Compile Meta
docOnlyMeta = Meta <$> (docPair <|> str <|> return Nothing) <*> pure []
  where
     docPair = symbol "@doc" >> str
```

Indeed this should be significantly faster than `meta`, which is nice to avoid when we don't need it.

Reply:
My thinking is that users will extrapolate that, since they can use `@model` on modules, schemas, and functions, they can use them on tables and consts as well. I think it's worth it to have a good error message in this case -- something like "@model not allowed in this declaration" rather than whatever the parser state happens to be.

Actually it appears that at the moment you can add anything to a `defconst` and it's silently accepted, which is even worse:

```
  (defconst AUTH_KEYSET 'K
    "Indicates keyset-governed account"
    sfkldjs
    sldkjflsdk
    slkdfjlsd)
```

which loads:

```
"Loaded module \"accounts\", hash \"ab906c26ee555aedd8d663bfd2f720ae065bc89e5f8addd51d0975525c295a8f60d900f697f0eb6a0e05793a94b91f2297b677a9ce947f329796b937849526e3\""
```

Reply:
That's a good catch on defconst, my guess is it's missing an `eof` if you want to try fixing that.

I suppose consts and schemas are relatively few, so if you want to go back to the boolean approach I think the perf. tradeoff is acceptable.

Reply:
I just updated this PR and fixed the `defconst` issue in #288.